ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/PartitionExport.java

```diff
@@ -115,11 +119,19 @@ void write(final ReplicationSpec forReplicationSpec) throws InterruptedException
                   .export(forReplicationSpec);
           LOG.debug("Thread: {}, finish partition dump {}", threadName, partitionName);
         } catch (Exception e) {
-          throw new RuntimeException("Error while export of data files", e);
+          throw new RuntimeException(e.getMessage(), e);
         }
-      });
+      }));
     }
     consumer.shutdown();
+    for (Future<?> future : futures) {
+      try {
+        future.get();
+      } catch (Exception e) {
+        LOG.error("failed", e.getCause());
+        throw new HiveException(e.getCause().getMessage(), e.getCause());
```

Comment:
Why do we need this special handling for PartitionExport with futures semantics? FileOperations.export itself take care of NPE when partition directory missing right?

Reply:
its done it a separate thread so it has to be done manually